libavformat/rtcenc.c

```diff
@@ -0,0 +1,2226 @@
+/*
+ * WebRTC-HTTP ingestion protocol (WHIP) muxer
+ * Copyright (c) 2023 The FFmpeg Project
+ *
+ * This file is part of FFmpeg.
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "config.h"
+
+#ifndef CONFIG_OPENSSL
+#error "DTLS is not supported, please enable openssl"
+#else
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+/**
+ * Minimum required version of OpenSSL.
+ *          MM NN FF PP S
+ * 0x1010102fL = 0x1 01 01 02 fL            // 1.1.1b release
+ *   MM(major) = 0x1                        // 1.*
+ *     NN(minor) = 0x01                     // 1.1.*
+ *          FF(fix) = 0x01                  // 1.1.1*
+ *     PP(patch) = 'a' + 0x02 - 1 = 'b'     // 1.1.1b *
+ *              S(status) = 0xf = release   // 1.1.1b release
+ * Status 0 for development, 1 to e for betas 1 to 14, and f for release.
+ * Please use the stable version for DTLS, see https://github.com/openssl/openssl/issues/346
+ */
+#if OPENSSL_VERSION_NUMBER < 0x100010b0L /* OpenSSL 1.0.1k */
+#error "OpenSSL version 1.0.1k or newer is required"
+#endif
+#endif
+
+#include "libavcodec/avcodec.h"
+#include "libavutil/base64.h"
+#include "libavutil/bprint.h"
+#include "libavutil/crc.h"
+#include "libavutil/hmac.h"
+#include "libavutil/opt.h"
+#include "libavutil/random_seed.h"
+#include "libavutil/time.h"
+#include "avc.h"
+#include "avio_internal.h"
+#include "http.h"
+#include "internal.h"
+#include "mux.h"
+#include "network.h"
+#include "srtp.h"
+
+/**
+ * Maximum size limit of a Session Description Protocol (SDP),
+ * be it an offer or answer.
+ */
+#define MAX_SDP_SIZE 8192
```

Comment:
Kieran Kunhya: 

```
?
```

Reply:
When generating an SDP offer or parsing an SDP answer, a maximum length for the SDP is established.